src/common/snippets/docs/snippets_design_guide.md

```diff
@@ -1,301 +1,465 @@
+Concluding this section, it's worth mentioning that the `LinearIR` currently provides two debug features: `debug_print()` and `serialize(...)`. The first one prints input and output `PortConnectors` and `PortDescriptors` for every `Expression` to stderr. The second one serializes the `LIR` into an `xml` OpenVINO graph format, where a lot of useful parameters are displayed for every `Expression`.
 
-##### Dialect extensibility
+##### Control flow optimization pipeline
 
-A target can potentially extend the snippets dialect with a target-specific operation for code emission. It should implement:
+Control flow optimization pipeline is divided into three main blocks: common pipeline, buffer pipeline and final pipeline. The common pipeline is mainly focused on an automatic loop injection and loop optimizations, but some transformations affecting data flow are also included. The exact set of transformations executed in the common pipeline will likely change as the `Snippets` evolve and develop, but it is worthwhile to cover some of them briefly to give you an idea on how the pipeline looks like:
+1. `MarkLoops` - performs an analysis of `Expressions'` connectivity and their `PortDescriptors`. Based on this information, the pass divides the `Expression` into groups, so that each of the groups can be executed inside one loop. Every group is described by a `loop_id`, and additional information is saved in `LinearIR::LoopManager::LoopInfo`.
+2. `FuseLoops` - analyzes the assigned `loop_ids` and `LoopInfo`, fuses some loops if possible. This pass can move some `Expressions` up or down the graph, so the `Expressions` with the same `loop_ids` are grouped together.
+3. `InsertBuffers` - analyzes `LoopInfo` and `Expression` semantics, inserts `snippets::op::Buffer`. `Buffer` is an operation that represents a memory buffer needed to save some intermediate results.
+4. `InsertLoadStore` - inserts explicit memory access operations like `Load` and `Store` (both from `snippets::op`). These are needed, so appropriate instructions will be emitted during the code generation stage to move data between memory and vector registers.
+5. `InitLoops` - inserts explicit `LoopBeegin` and `LoopEnd` (`snippets::op`) operations based on the acquired `LoopInfo`. Again, the explicit operations are needed to emit appropriate instructions later.
+6. `LoadMoveBroadcastToBroadcastLoad` fuse `Load->MoveBroadcast` sequences into a single `BroadcastLoad` expressions.
 
-* nGraph operation (for example, `class FMA : public ov::op::Op`)
-* Emitter for the operation (for example, `class FmaEmitter : public Emitter` )
-* register the pair in `jitters` map
+The buffer pipeline is focused on managing `op::Buffer` operations:
+1. `IdentifyBuffers` - analyzes buffers using graph coloring algorithm to enable memory reuse.
+2. `AllocateBuffers` - calculates minimal memory size required to handle all the buffers. The memory will be allocated during the execution stage by the plugin `Snippet` node.
```

Comment:
Maybe we should notice that in the near future we are going to write whole `BufferManager` which calculates memory size for allocation using `MemorySolver` and marks `Buffers` which can be inplace instead of `AllocateBuffers` pass. What do you think?

Reply:
The purpose of this documentation is to make it easier for the developers to navigate the Snippets pipeline, so the docs should reflect the actual state of the pipeline.
That said, I think that our plans should be in presentations and jira tickets rather than in the docs.